styler.py

```diff
@@ -0,0 +1,218 @@
+# matplotlib parameters for publication plots
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+from matplotlib import gridspec
+import re
+import os
+
+this_folder = os.path.dirname(__file__)
+
+
+#### styler should also provide colors for consistency!
+## are these ok?
+## should actually be a dictionary with sane keys...
+colors = ['#4878cf', '#6acc65', '#d65f5f', '#b47cc7', '#c4ad66', '#77bedb']
+
+
+params = {'text.latex.preamble': [r'\usepackage{lmodern}',
+                                  r'\usepackage{amsfonts,amsmath,amssymb}',
+                                  r'\mathchardef\mhyphen="2D',
+                                  r'\DeclareMathOperator{\ms}{m\!\cdot\!s^{-1}}',
+                                  r'\DeclareMathOperator{\kms}{km\!\cdot\!s^{-1}}'],
+          'text.usetex' : True,
+          'font.size'   : 8,
+          'font.family' : 'lmodern',
+          'text.latex.unicode': True,
+          'axes.unicode_minus': True,
+          }
+
+class MyFormatter(ticker.ScalarFormatter):
+    """ Axis ticks formatter to replace the big minus sign with a smaller, prettier one. """
+    def __call__(self, x, pos=None):
+        # call the original ScalarFormatter
+        rv = ticker.ScalarFormatter.__call__(self, x, pos)
+        # check if we really use TeX
+        if plt.rcParams["text.usetex"]:
+            # if we have the string ^{- there is a negative exponent
+            # where the minus sign is replaced by the short hyphen
+            rv = re.sub(r'-', r'\mhyphen', rv)
+
+        rv = rv.replace('.0', '')
```

Comment:
This might remove unwanted `.0`.

    if rv.endswith('.0'):
        rv = rv.replace('.0', '')

or something similar